src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,239 @@
+                similarityScoreEntity.BaseVersionType = ResourceContentVersionTypes.Base;
+                similarityScoreEntity.ComparedVersionType = ResourceContentVersionTypes.Snapshot;
+
+                similarityScoreEntity.ComparedVersionId = message.CompareVersionId 
+                                                          ?? throw new InvalidOperationException(
+                                                              $"Snapshot version {message.BaseVersionId} not found.");
+                
+                baseVersionText = await GetBaseVersionContentText(similarityScoreEntity.BaseVersionId, dbContext, ct);
+                compareVersionText = await GetSnapshotVersionText(similarityScoreEntity.ComparedVersionId, dbContext, ct);
+                
+                _logger.LogInformation(
+                    "Scoring resource content version {BaseVersionId} vs snapshot version {CompareVersionId}...",
+                    similarityScoreEntity.BaseVersionId,
+                    similarityScoreEntity.ComparedVersionId);
+
+                break;
+            case ResourceContentVersionSimilarityComparisonType.SnapshotToSnapshot:
+                similarityScoreEntity.BaseVersionType = ResourceContentVersionTypes.Snapshot;
+                similarityScoreEntity.ComparedVersionType = ResourceContentVersionTypes.Snapshot;
+
+                similarityScoreEntity.ComparedVersionId = message.CompareVersionId 
+                                                          ?? throw new InvalidOperationException(
+                                                              $"Snapshot version {message.CompareVersionId} not found.");
+                
+                baseVersionText = await GetSnapshotVersionText(similarityScoreEntity.BaseVersionId, dbContext, ct);
+                compareVersionText = await GetSnapshotVersionText(similarityScoreEntity.ComparedVersionId, dbContext, ct);
+                
+                _logger.LogInformation(
+                    "Scoring snapshot version {BaseVersionId} vs snapshot version {CompareVersionId}...",
+                    similarityScoreEntity.BaseVersionId,
+                    similarityScoreEntity.ComparedVersionId);
+
+                break;
+            default:
+                throw new InvalidOperationException($"Unknown similarity comparison type: {message.ComparisonType}");
+        }
+        
+        similarityScoreEntity.SimilarityScore = StringSimilarityUtilities.ComputeLevenshteinSimilarity(baseVersionText, compareVersionText);
+
+        return similarityScoreEntity;
+    }
+    
+    private static async Task<ResourceContentVersionMachineTranslationEntity> GetMachineTranslationById(
+        int machineTranslationId,
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        return await dbContext
+                .ResourceContentVersionMachineTranslations
+                .AsTracking()
```

Comment:
It looks like none of the EF queries in this class need `AsTracking()` because none of the entities retrieved are used in any DB write operations.